src/geff/interops/trackmate_xml.py

```diff
@@ -462,55 +482,450 @@ def _parse_model_tag(
+            None will be used.
+    """
+    # Dictionaries of metadata for GEFF properties.
+    node_props_metadata = {}  # type: dict[str, Any]
+    edge_props_metadata = {}  # type: dict[str, Any]
+    lineage_props_metadata = {}  # type: dict[str, Any]
+    props_metadata = {
+        "node_props_metadata": node_props_metadata,
+        "edge_props_metadata": edge_props_metadata,
+        "lineage_props_metadata": lineage_props_metadata,
+    }
+    # Mapping TrackMate feature types to GEFF metadata fields.
+    mapping_feat_type = {
+        "SpotFeatures": node_props_metadata,
+        "EdgeFeatures": edge_props_metadata,
+        "TrackFeatures": lineage_props_metadata,
+    }
+
+    tags_data = _get_specific_tags(xml_path, ["FeatureDeclarations"])
+    xml_md = tags_data["FeatureDeclarations"]
+    for feat_type in xml_md:
+        if feat_type.tag in mapping_feat_type:
+            for feat in feat_type.findall("Feature"):
+                _process_feature_metadata(
+                    feat, mapping_feat_type[feat_type.tag], feat_type.tag, units
+                )
+
+    return props_metadata
+
+
+def _build_geff_metadata(
+    xml_path: Path,
+    units: dict[str, str],
+    img_path: str | None,
+    trackmate_metadata: dict[str, ET.Element],
+    props_metadata: dict[str, dict[str, Any]],
+) -> GeffMetadata:
+    """Create GEFF metadata from TrackMate XML data.
+
+    Args:
+        xml_path (Path): The path to the TrackMate XML file.
+        units (dict[str, str]): A dictionary containing the units of the model.
+        img_path (str | None): The path to the related image file.
+        trackmate_metadata (dict[str, dict[str, ET.Element]]): The TrackMate metadata extracted
+            from the XML file.
+        props_metadata (dict[str, Any]): The properties metadata extracted from the XML file.
+
+    Returns:
+        GeffMetadata: The constructed GEFF metadata object.
+    """
+    extra = {
+        "trackmate_version": _get_trackmate_version(xml_path),
+        "provenance": "trackmate",
+        "lineage_props_metadata": props_metadata["lineage_props_metadata"],
+        "trackmate": {
+            "log": ET.tostring(trackmate_metadata["Log"], encoding="utf-8").decode(),
+            "settings": ET.tostring(trackmate_metadata["Settings"], encoding="utf-8").decode(),
+            "gui_state": ET.tostring(trackmate_metadata["GUIState"], encoding="utf-8").decode(),
+            "display_settings": ET.tostring(
+                trackmate_metadata["DisplaySettings"], encoding="utf-8"
+            ).decode(),
+        },
+    }
+
+    return GeffMetadata(
+        axes=[
+            Axis(name="POSITION_X", type="space", unit=units.get("spatialunits", "pixel")),
+            Axis(name="POSITION_Y", type="space", unit=units.get("spatialunits", "pixel")),
+            Axis(name="POSITION_Z", type="space", unit=units.get("spatialunits", "pixel")),
```

Comment:
Mostly a curiosity question, but does trackmate always assume a z dimension? How would 2d data be handled?